data-plane/dispatcher/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/ConsumerDeployer.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright 2020 The Knative Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package dev.knative.eventing.kafka.broker.dispatcher;
+
+import static net.logstash.logback.argument.StructuredArguments.keyValue;
+
+import dev.knative.eventing.kafka.broker.contract.DataPlaneContract;
+import dev.knative.eventing.kafka.broker.core.reconciler.EgressReconcilerListener;
+import io.vertx.core.Future;
+import io.vertx.core.Vertx;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * ResourcesManager manages Resource and Egress objects by instantiating and starting verticles based on resources
+ * configurations.
+ *
+ * <p>Note: {@link ConsumerDeployer} is not thread-safe and it's not supposed to be shared between
+ * threads.
+ */
+public final class ConsumerDeployer implements EgressReconcilerListener {
+
+  private static final Logger logger = LoggerFactory.getLogger(ConsumerDeployer.class);
+
+  private final Map<String, String> deployedDispatchers;
+
+  private final Vertx vertx;
+  private final ConsumerVerticleFactory consumerFactory;
+
+  /**
+   * All args constructor.
+   *
+   * @param vertx                   vertx instance.
+   * @param consumerFactory         consumer factory.
+   * @param egressesInitialCapacity egresses container initial capacity.
+   */
+  public ConsumerDeployer(
+    final Vertx vertx,
+    final ConsumerVerticleFactory consumerFactory,
+    final int egressesInitialCapacity) {
+    Objects.requireNonNull(vertx, "provide vertx instance");
+    Objects.requireNonNull(consumerFactory, "provide consumer factory");
+    if (egressesInitialCapacity <= 0) {
+      throw new IllegalArgumentException("egressesInitialCapacity cannot be negative or 0");
+    }
+    this.vertx = vertx;
+    this.consumerFactory = consumerFactory;
+    this.deployedDispatchers = new HashMap<>(egressesInitialCapacity);
```

Comment:
Is it safe to use a `HashMap` since `deployVerticle` is async?

Reply:
The threading model is not modified  in this pr... Although, i'm gonna investigate this a bit further

Reply:
Before we added it here https://github.com/knative-sandbox/eventing-kafka-broker/pull/327/files#diff-0f8e03ae346d693e18b8ff2c5362f82d68de2449ec3f695bda06b54e754c2f53L173, now it's here https://github.com/knative-sandbox/eventing-kafka-broker/pull/327/files#diff-16e3053ac32fc1b4274a27aa8f7f76ca6195102fc139937f6903a8e9fce3d97bR83